cluster/ci/origin/logging-infra/gcs.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"sync"
+
+	"cloud.google.com/go/storage"
+)
+
+// Metadata holds run-time data about the build
+type Metadata struct {
+	// JobName is the name of the job triggering
+	// this build
+	JobName string `json:"job-name"`
+
+	// BuildNumber is the identifier for this build
+	// of the job
+	BuildNumber int `json:"build-number"`
+
+	// Parameters contains all parameters passed
+	// to this build
+	Parameters map[string]string `json:"parameters"`
+}
+
+// GcsPath determines where artifacts for this job should
+// be uploaded to under the base bucket
+func (m *Metadata) GcsPath() string {
```

Comment:
Right, storing just by `{name}/{build}/` and having the display server figure out how to thread through is my favored approach as well.